rmw_connextdds_common/src/common/rmw_type_support.cpp

```diff
@@ -830,7 +830,8 @@ void RMW_Connext_MessageTypeSupport::type_info(
 
   if (keyed) {
     key_callbacks = *callbacks->key_callbacks;
-    key_serialized_size_max = key_callbacks.max_serialized_size_key(unbounded_key);
+    key_serialized_size_max = static_cast<uint32_t>(
+      key_callbacks.max_serialized_size_key(unbounded_key));
```

Comment:
@fgallegosalido 

is it possible that `max_serialized_size_key` returns the bigger value than 32bit unsigned int? if so, the truncated value may lead to incorrect calculations or behavior  below. Or `max_serialized_size_key` guarantees to return the value in the range of 32bit unsigned int?

Reply:
It would probably be very unlikely that a type has a bounded size bigger than 32 bits, but it would be a reasonable check to do. Although, the check should be against `RTI_CDR_MAX_SERIALIZED_SIZE`. Otherwise, return an error or something

Reply:
What's the suggestion ? Add a check before the cast and generate and error ?

Reply:
Yeah, check if `key_callbacks.max_serialized_size_key(unbounded_key) < RTI_CDR_MAX_SERIALIZED_SIZE`. Otherwise return an error/throw exception, whichever fits better.